CMakeLists.txt

```diff
@@ -387,7 +387,18 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 if(CMAKE_SYSTEM_NAME MATCHES "Darwin")
   if(ON_APPLE_M1)
-    list(APPEND CMAKE_PREFIX_PATH "/opt/homebrew/opt/icu4c")
+    execute_process(
+      COMMAND brew --prefix icu4c
+      RESULT_VARIABLE BREW_ICU4C
+      OUTPUT_VARIABLE BREW_ICU4C_PREFIX
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+    if(BREW_ICU4C EQUAL 0 AND EXISTS "${BREW_ICU4C_PREFIX}")
+      message(
+        STATUS "Found icu4c installed by Homebrew at ${BREW_ICU4C_PREFIX}")
+      list(APPEND CMAKE_PREFIX_PATH "${BREW_ICU4C_PREFIX}")
```

Comment:
I think we should move this before `ON_APPLE_M1`  so it's used for both intel and m1 mac and then only distinguish arch's for fallback paths.

Reply:
cc: @tanjialiang

Reply:
Saw this after landed. I can have another followup PR to address this. Thanks!